### DISCUSSION TITLE
Is there multicall/multicall2 in wagmi?

### DISCUSSION THREAD

**Discussion by nguyenhuugiatri** [wevm/wagmi#1512] [Q&A]
I found that Wagmi is supporting multicall3.
In my case, the chain I'm using is still using multicall2, I wonder is there any way to integrate multicall2 with Wagmi?

Thank for your support <3

**Comment by tmm** [comment]
The [multicall3](https://github.com/mds1/multicall) contract we use is backwards compatible with multicall2 and multicall so you should be able to pop in a multicall2 address for the `multicall3` key in [`Chain`](https://wagmi.sh/react/chains#build-your-own).

**Reply by nguyenhuugiatri** [reply]
Thank @tmm for your reply.
It looks like Wagmi uses `aggregate3` which multicall3 only supports.
Did I miss something?

<img width="730" alt="image" src="https://user-images.githubusercontent.com/44458003/208163728-42a5a1eb-9724-4ff8-801f-f8fe265835aa.png">
<img width="551" alt="image" src="https://user-images.githubusercontent.com/44458003/208163771-d953c796-3bce-428f-8be0-6042ba28c1a3.png">

**Reply by tmm** [reply]
Nope. We need to add in support for backwards compatibility 😅 What chain are you using?

**Reply by nguyenhuugiatri** [reply]
I'm using [Ronin](https://roninchain.com/) chain.

Multicall2 contracts:
Mainnet: [0xc76d0d0d3aa608190f78db02bf2f5aef374fc0b9](https://explorer.roninchain.com/address/ronin:c76d0d0d3aa608190f78db02bf2f5aef374fc0b9)
Testnet: [0x31c9ef8a631e2489e69833df3b2cb4bf0dc413bc](https://saigon-explorer.roninchain.com/address/ronin:31c9ef8a631e2489e69833df3b2cb4bf0dc413bc)

**Reply by mds1** [reply]
Hey, Multicall3 author here. If you need Multicall3 deployed to ronin (or any other chains), just send some funds to cover the deploy cost to the deployer account: `0x05f32B3cC3888453ff71B01135B34FF8e41263F2`

(Getting deployment funds across so many chains is tedious, so deployment can happen much sooner if someone funds the deployer account, and I'm happy to reimburse on another chain if you need)
